### PR TITLE
Update email alerting provider to supply a username

### PR DESCRIPTION
**Pull request by tmoitie**
This is intended to maintain backwards compatibility with using Provider.from.

I am unfamiliar with golang so apologies for any style mistakes

### CODE REVIEW

**Comment by TwiN** [comment]
Just so I can understand; this enables using a username that is not the same as the one provided in `alerting.email.from` to authenticate while allowing the usage of `alerting.email.from` as a sender?

**Comment by tmoitie** [comment]
Yep, that's exactly what I intended

File: README.md [hunk 1]
```diff
@@ -352,20 +352,22 @@ endpoints:
...
-| `alerting.email.port`          | Port the mail server is listening to (e.g. `587`)                                          | Required `0`  |
-| `alerting.email.to`            | Email(s) to send the alerts to                                                             | Required `""` |
-| `alerting.email.default-alert` | Default alert configuration. <br />See [Setting a default alert](#setting-a-default-alert) | N/A           |
+| Parameter                      | Description                                                                                | Default               |
+|:-------------------------------|:-------------------------------------------------------------------------------------------|:----------------------|
+| `alerting.email`               | Configuration for alerts of type `email`                                                   | `{}`                  |
+| `alerting.email.from`          | Email used to send the alert                                                               | Required `""`         |
+| `alerting.email.username`      | Username of the SMTP server used to send the alert                                         | `alerting.email.from` |
```
Comment (TwiN) [inline README.md:359]:
```suggestion
| `alerting.email.username`      | Username of the SMTP server used to send the alert. If empty, uses `alerting.email.from`.  | `""`                  |
```

**Review by TwiN — CHANGES_REQUESTED** [review]
Just a small change, and you're good to go!

**Comment by TwiN** [comment]
@tmoitie Thank you for your contribution!

**Pull request merged**